NeuralNetwork/NeatNN.js

```diff
@@ -139,55 +141,59 @@ class NeatNN {
     *  Penalty is calculated by the number of connections and nodes
     */ 
     CalculatePenalty() {
-        this.penalty = this.connections.length * 0.005 + this.nodes.length * 0.02;
+        this.penalty = this.connections.length * 0.002 + this.nodes.length * 0.005;
     }
 
-    // ---------------------------- Mutate Functions ------------------------------------
+    // ---------------------------- Mutate Functions ------------------------------------ //
 
     // Mutate the network
     Mutate() {
 
-        var mutateWeightChance = 0.6;
-        var mutateBiasChance = 0.3;
-        var mutateAddConnectionChance = 0.06;
-        var mutateRemoveConnectionChance = 0.05;
-        var mutateAddNodeChance = 0.05;
-        var mutateRemoveConnectionChance = 0.04;
+        var mutateWeightChance =            0.6  * NeatNN.MUTATION_MULTIPLIER;
+        var mutateBiasChance =              0.3  * NeatNN.MUTATION_MULTIPLIER;
+        var mutateAddConnectionChance =     0.05 * NeatNN.MUTATION_MULTIPLIER;
+        var mutateRemoveConnectionChance =  0.05 * NeatNN.MUTATION_MULTIPLIER;
+        var mutateAddNodeChance =           0.03 * NeatNN.MUTATION_MULTIPLIER;
+        var mutateRemoveConnectionChance =  0.03 * NeatNN.MUTATION_MULTIPLIER;
 
         // Mutate the weight of a random connection
-        if (Math.random() < mutateWeightChance){
+        while (mutateWeightChance > 0 && Math.random() < mutateWeightChance){
+            mutateWeightChance -= 1;
```

Comment:
why are the mutation chances being decremented?

Reply:
They are local variables, so it is only for this function call. They are in a while loop so if you increase NeatNN.MUTATION_MULTIPLIER, the mutation rate can be above 100% and still work. By default, if they are above 100%, it will mutate, subtract 100%, and try again.